### PR TITLE
MINOR: Add missing generics and surpress warning annotations

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
@bbejeck 

This should go into 1.1 (and maybe 1.0), too.

File: streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java [hunk 1]
```diff
@@ -584,6 +584,7 @@ public synchronized boolean hasNext() {
             this.rocksDBStore = rocksDBStore;
         }
 
+        @SuppressWarnings("unchecked")
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java]: Why do we need this? Seems no unchecked warnings?

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java]: IntelliJ does not show a warning, but `./gradlew` does -- I am also not sure why this warning appears in the build to be honest... It seems to be type safe code to me.

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java]: yeah same here.. could you paste the warning message here if you still have it convienent?

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java]:
```
/Users/matthias/IdeaProjects/kafka/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java:589: warning: [unchecked] unchecked call to restoreAllInternal(Collection<KeyValue<byte[],byte[]>>) as a member of the raw type RocksDBStore
            rocksDBStore.restoreAllInternal(records);
                                           ^
```

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java]: Maybe this is the root cause: a few lines above:

```
private final RocksDBStore rocksDBStore;
```

change to 

```
private final RocksDBStore<K, V> rocksDBStore;
```

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java]: Good catch. This change implies a lot of other changes, too... I added a commit with all those.

**Comment by mjsax** [comment]
Updated this.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/310/

**Comment by dguy** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/325/

**Comment by mjsax** [comment]
Retest this please.

**Review by bbejeck — APPROVED** [review]
Thanks @mjsax  LGTM

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/339/

File: streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java [hunk 2]
```diff
@@ -311,6 +311,7 @@ public void onRestoreEnd(TopicPartition topicPartition, String storeName, long t
      *
      * @param consumerRecord the record to be processed
      */
+    @SuppressWarnings("deprecation")
```
Comment (guozhangwang) [inline streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java]: Not clear which callee is deprecated?

**Review by guozhangwang — COMMENTED** [review]
One comment, otherwise lgtm. Great work reducing RocksDBStore's templates!

File: streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java (hunk 2 above)
Comment (mjsax) [reply streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java]: It's `consumerRecord.checksum()` -- but we can't annotate within the method, can we?

Should I add a comment instead or do you want the annotation be removed?

Comment (guozhangwang) [reply streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java]: Could we just use `NULL_CHECKSUM`?

Comment (mjsax) [reply streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java]: Well. Yes and no. We actually just pass the checksum from the input `ConsumerRecord` -- thus, we would loose this information. However, I guess nobody should use checksum and the maybe we don't need to pass the checksum necessarily?

Comment (guozhangwang) [reply streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java]: Yup, if people mistakenly validate checksums that is not encouraged as in the deprecation message, so I think we should be fine.

Comment (mjsax) [reply streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java]: Ack.

**Comment by mjsax** [comment]
Updated this.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/363/

**Comment by guozhangwang** [comment]
Findbugs failed: 

```
FindBugs rule violations were found. See the report at: file:///home/jenkins/jenkins-slave/workspace/kafka-pr-jdk8-scala2.12/streams/build/reports/findbugs/main.xml
03:28:12 
```

**Comment by dguy** [comment]
retest this please

**Comment by mjsax** [comment]
Updated and rebase. Also did some more cleanup.

File: streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java [hunk 3]
```diff
@@ -238,31 +221,30 @@ private void validateStoreOpen() {
...
 
-    private byte[] getInternal(byte[] rawKey) {
+    private byte[] getInternal(final byte[] rawKey) {
         try {
             return this.db.get(rawKey);
-        } catch (RocksDBException e) {
-            throw new ProcessorStateException("Error while getting value for key " + serdes.keyFrom(rawKey) +
-                    " from store " + this.name, e);
+        } catch (final RocksDBException e) {
+            throw new ProcessorStateException("Error while getting value for key from store " + this.name, e);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java:228]: This change is intentional. Same below removing raw key/value from the exception error message. Cf. https://issues.apache.org/jira/browse/KAFKA-6538 as follow up for this change.

**Review by guozhangwang — APPROVED** [review]

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `1.1` branch.

**Pull request merged**